tinygrad/ops.py

```diff
@@ -164,6 +165,13 @@ def __init__(self, jit_cache: List[JitItem], input_rawbuffers: List[RawBuffer],
         if a in input_rawbuffers:
           self.input_replace[(j,i)] = input_rawbuffers.index(a)
     assert len(set(self.input_replace.values())) == len(input_rawbuffers), "some input tensors not found"
+
+    # Record indices of jit items components of which could be updated each run.
+    def dim_has_vars(dim): return dim and not all_int(dim)
```

Comment:
Isn't just checking not all_int enough?

Reply:
Backends without locals could have None here, like LLVM